Userland/Utilities/ps.cpp

```diff
@@ -20,14 +20,25 @@ static ErrorOr<String> determine_tty_pseudo_name()
         perror("fstat");
         return Error::from_errno(saved_errno);
     }
+
     int tty_device_major = major(tty_stat.st_rdev);
     int tty_device_minor = minor(tty_stat.st_rdev);
+
     if (tty_device_major == 201) {
         return String::formatted("pts:{}", tty_device_minor);
     }
+
     if (tty_device_major == 4) {
         return String::formatted("tty:{}", tty_device_minor);
     }
+
+    // Handle the /dev/null memory device.
+    // Not doing so, will give an assertion failure
+    // when forwarding /dev/null into ps.
+    if (tty_device_major == 1 && tty_device_minor == 3) {
+        return "null";
+    }
+
     VERIFY_NOT_REACHED();
```

Comment:
Okay but what if I do `ps < /dev/zero` now? :^)

I suggest we fix this for any non-TTY by simply returning `"n/a"` instead of the `VERIFY_NOT_REACHED()`

Reply:
Can do that. I did not know if there was a reason for the assertion. :-)